### PR TITLE
Move Parse object manipulation to client side

### PR DESCRIPTION
**Pull request by nathanathan**
This is being done because the parse user is only logged in client side so Parse calls on the server side can't use access control.

### CODE REVIEW

File: app/packages/openam/login.coffee [hunk 1]
```diff
@@ -8,4 +8,9 @@ Template.login.events
 
     Meteor.call 'loginUser', email, passw, (err, token) ->
       Meteor.loginWithToken(token)
-      Parse.User.logIn email, passw
+      Parse.User.logIn(email, passw)
+        .then ()->
+          console.log(Parse.User.current())
+        .fail (error)->
+          toastr.error('Could not log into Parse.')
+          console.log(error)
```
Comment (arnoldbm) [inline app/packages/openam/login.coffee]: We should roll back the meteor login if the parse login fails. Could make for hard to debug behavior otherwise.

**Comment by dan-nyanko** [comment]
I cannot create a new survey

image

**Comment by jgoley** [comment]
@dan-nyanko I think it may be related to the latest commit where I changed createdBy to a pointer to the Parse User. I think if you pull the latest, survey creation works.

File: app/imports/models.coffee [hunk 2]
```diff
@@ -149,21 +133,22 @@ Trigger = Parse.Object.extend 'Trigger',
   create: (props, form) ->
     @setProperties props
     @save()
-      .then (trigger) =>
+      .then =>
         @addToForm(form)
       .then ->
-        trigger.id
+        @id
```
Comment (nathanathan) [inline app/imports/models.coffee]: @jgoley `@id` is not in a fat arrow function here so it might have the wrong context.

**Comment by jgoley** [comment]
@dan-nyanko @nathanathan I think I've fixed everything now. I refactored things a little more—now we get the last question order before calling the create method (thought it made sense to keep it in the related class).

**Pull request merged**